src/psyclone/domain/common/extract_driver_creator.py

```diff
@@ -598,8 +622,90 @@ def create(self, nodes, input_list, output_list,
 
         self.add_result_tests(program, output_symbols)
 
-        code = writer(file_container)
+        return file_container
+
+    # -------------------------------------------------------------------------
+    def get_driver_as_string(self, nodes, input_list, output_list,
+                             prefix, postfix, region_name,
+                             writer=FortranWriter()):
+        # pylint: disable=too-many-arguments
+        '''This function uses 'create()` function to get a PSyIR of a
+        stand-alone driver, and then uses the provided language writer
+        to create a string representation in the selected language
+        (defaults to Fortran).
+
+        :param nodes: a list of nodes.
+        :type nodes: list of :py:obj:`psyclone.psyir.nodes.Node`
+        :param input_list: list of variables that are input parameters.
+        :type input_list: list of :py:class:`psyclone.core.Signature`
+        :param output_list: list of variables that are output parameters.
+        :type output_list: list or :py:class:`psyclone.core.Signature`
+        :param str prefix: the prefix to use for each PSyData symbol, \
+            e.g. 'extract' as prefix will create symbols `extract_psydata`.
+        :param str postfix: a postfix that is appended to an output variable \
+            to create the corresponding variable that stores the output \
+            value from the kernel data file. The caller must guarantee that \
+            no name clashes are created when adding the postfix to a variable \
+            and that the postfix is consistent between extract code and \
+            driver code (see 'ExtractTrans.determine_postfix()').
+        :param (str,str) region_name: an optional name to \
+            use for this PSyData area, provided as a 2-tuple containing a \
+            location name followed by a local name. The pair of strings \
+            should uniquely identify a region.
+        :param language_writer: a backend visitor to convert PSyIR \
+            representation to the selected language. It defaults to \
+            the FortranWriter.
+        :type language_writer: an optional instance of \
+            :py:class:`psyclone.psyir.backend.language_writer.LanguageWriter`
+
+        :return: the string representation of the driver in the selected \
```

Comment:
s/return/returns/ and s/the string representation of//

Reply:
I've opened #1402 to update our coding style (:return: is correct and frequently used, though :returns: much more often). I cover the ``()`` for multi line, :return:", and new testing comments.

Reply:
Thanks Joerg.